{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "019c47fb-b014-490a-b280-33af9cd5ca44",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fetching data for 2005-02-01...\n",
      "Error fetching data for 2005-02-01: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-01 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-02...\n",
      "Error fetching data for 2005-02-02: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-02 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-03...\n",
      "Error fetching data for 2005-02-03: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-03 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-04...\n",
      "Error fetching data for 2005-02-04: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-04 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-05...\n",
      "Error fetching data for 2005-02-05: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-05 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-06...\n",
      "Error fetching data for 2005-02-06: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-06 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-07...\n",
      "Error fetching data for 2005-02-07: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-07 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-08...\n",
      "Error fetching data for 2005-02-08: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-08 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-09...\n",
      "Error fetching data for 2005-02-09: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-09 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-10...\n",
      "Error fetching data for 2005-02-10: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-10 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-11...\n",
      "Error fetching data for 2005-02-11: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-11 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-12...\n",
      "Error fetching data for 2005-02-12: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-12 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-13...\n",
      "Error fetching data for 2005-02-13: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-13 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-14...\n",
      "Error fetching data for 2005-02-14: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-14 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-15...\n",
      "Error fetching data for 2005-02-15: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-15 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-16...\n",
      "Error fetching data for 2005-02-16: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-16 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-17...\n",
      "Error fetching data for 2005-02-17: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-17 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-18...\n",
      "Error fetching data for 2005-02-18: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-18 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-19...\n",
      "Error fetching data for 2005-02-19: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-19 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-20...\n",
      "Error fetching data for 2005-02-20: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-20 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-21...\n",
      "Error fetching data for 2005-02-21: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-21 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-22...\n",
      "Error fetching data for 2005-02-22: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-22 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-23...\n",
      "Error fetching data for 2005-02-23: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-23 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-24...\n",
      "Error fetching data for 2005-02-24: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-24 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-25...\n",
      "Error fetching data for 2005-02-25: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-25 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-26...\n",
      "Error fetching data for 2005-02-26: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-26 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-27...\n",
      "Error fetching data for 2005-02-27: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-27 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-28...\n",
      "Error fetching data for 2005-02-28: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-28 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-01...\n",
      "Error fetching data for 2005-03-01: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-01 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-02...\n",
      "Error fetching data for 2005-03-02: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-02 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-03...\n",
      "Error fetching data for 2005-03-03: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-03 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-04...\n",
      "Error fetching data for 2005-03-04: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-04 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-05...\n",
      "Error fetching data for 2005-03-05: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-05 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-06...\n",
      "Error fetching data for 2005-03-06: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-06 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-07...\n",
      "Error fetching data for 2005-03-07: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-07 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-08...\n",
      "Error fetching data for 2005-03-08: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-08 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-09...\n",
      "Error fetching data for 2005-03-09: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-09 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-10...\n",
      "Error fetching data for 2005-03-10: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-10 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-11...\n",
      "Error fetching data for 2005-03-11: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-11 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-12...\n",
      "Error fetching data for 2005-03-12: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-12 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-13...\n",
      "Error fetching data for 2005-03-13: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-13 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-14...\n",
      "Error fetching data for 2005-03-14: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-14 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-15...\n",
      "Error fetching data for 2005-03-15: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-15 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-16...\n",
      "Error fetching data for 2005-03-16: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-16 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-17...\n",
      "Error fetching data for 2005-03-17: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-17 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-18...\n",
      "Error fetching data for 2005-03-18: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-18 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-19...\n",
      "Error fetching data for 2005-03-19: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-19 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-20...\n",
      "Error fetching data for 2005-03-20: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-20 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-21...\n",
      "Error fetching data for 2005-03-21: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-21 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-22...\n",
      "Error fetching data for 2005-03-22: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-22 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-23...\n",
      "Error fetching data for 2005-03-23: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-23 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-24...\n",
      "Error fetching data for 2005-03-24: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-24 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-25...\n",
      "Error fetching data for 2005-03-25: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-25 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-26...\n",
      "Error fetching data for 2005-03-26: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-26 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-27...\n",
      "Error fetching data for 2005-03-27: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-27 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-28...\n",
      "Error fetching data for 2005-03-28: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-28 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-29...\n",
      "Error fetching data for 2005-03-29: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-29 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-30...\n",
      "Error fetching data for 2005-03-30: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-30 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-31...\n",
      "Error fetching data for 2005-03-31: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-31 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-01...\n",
      "Error fetching data for 2005-04-01: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-01 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-02...\n",
      "Error fetching data for 2005-04-02: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-02 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-03...\n",
      "Error fetching data for 2005-04-03: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-03 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-04...\n",
      "Error fetching data for 2005-04-04: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-04 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-05...\n",
      "Error fetching data for 2005-04-05: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-05 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-06...\n",
      "Error fetching data for 2005-04-06: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-06 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-07...\n",
      "Error fetching data for 2005-04-07: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-07 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-08...\n",
      "Error fetching data for 2005-04-08: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-08 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-09...\n",
      "Error fetching data for 2005-04-09: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-09 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-10...\n",
      "Error fetching data for 2005-04-10: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-10 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-11...\n",
      "Error fetching data for 2005-04-11: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-11 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-12...\n",
      "Error fetching data for 2005-04-12: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-12 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-13...\n",
      "Error fetching data for 2005-04-13: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-13 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-14...\n",
      "Error fetching data for 2005-04-14: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-14 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-15...\n",
      "Error fetching data for 2005-04-15: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-15 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-16...\n",
      "Error fetching data for 2005-04-16: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-16 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-17...\n",
      "Error fetching data for 2005-04-17: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-17 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-18...\n",
      "Error fetching data for 2005-04-18: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-18 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-19...\n",
      "Error fetching data for 2005-04-19: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-19 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-20...\n",
      "Error fetching data for 2005-04-20: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-20 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-21...\n",
      "Error fetching data for 2005-04-21: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-21 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-22...\n",
      "Error fetching data for 2005-04-22: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-22 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-23...\n",
      "Error fetching data for 2005-04-23: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-23 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-24...\n",
      "Error fetching data for 2005-04-24: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-24 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-25...\n",
      "Error fetching data for 2005-04-25: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-25 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-26...\n",
      "Error fetching data for 2005-04-26: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-26 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-27...\n",
      "Error fetching data for 2005-04-27: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-27 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-28...\n",
      "Error fetching data for 2005-04-28: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-28 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-29...\n",
      "Error fetching data for 2005-04-29: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-29 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-04-30...\n",
      "Error fetching data for 2005-04-30: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-04-30 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-01...\n",
      "Error fetching data for 2005-05-01: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-01 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-02...\n",
      "Error fetching data for 2005-05-02: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-02 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-03...\n",
      "Error fetching data for 2005-05-03: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-03 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-04...\n",
      "Error fetching data for 2005-05-04: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-04 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-05...\n",
      "Error fetching data for 2005-05-05: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-05 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-06...\n",
      "Error fetching data for 2005-05-06: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-06 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-07...\n",
      "Error fetching data for 2005-05-07: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-07 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-08...\n",
      "Error fetching data for 2005-05-08: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-08 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-09...\n",
      "Error fetching data for 2005-05-09: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-09 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-10...\n",
      "Error fetching data for 2005-05-10: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-10 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-11...\n",
      "Error fetching data for 2005-05-11: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-11 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-12...\n",
      "Error fetching data for 2005-05-12: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-12 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-13...\n",
      "Error fetching data for 2005-05-13: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-13 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-14...\n",
      "Error fetching data for 2005-05-14: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-14 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-15...\n",
      "Error fetching data for 2005-05-15: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-15 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-16...\n",
      "Error fetching data for 2005-05-16: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-16 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-17...\n",
      "Error fetching data for 2005-05-17: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-17 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-18...\n",
      "Error fetching data for 2005-05-18: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-18 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-19...\n",
      "Error fetching data for 2005-05-19: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-19 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-20...\n",
      "Error fetching data for 2005-05-20: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-20 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-21...\n",
      "Error fetching data for 2005-05-21: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-21 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-22...\n",
      "Error fetching data for 2005-05-22: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-22 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-23...\n",
      "Error fetching data for 2005-05-23: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-23 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-24...\n",
      "Error fetching data for 2005-05-24: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-24 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-25...\n",
      "Error fetching data for 2005-05-25: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-25 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-26...\n",
      "Error fetching data for 2005-05-26: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-26 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-27...\n",
      "Error fetching data for 2005-05-27: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-27 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-28...\n",
      "Error fetching data for 2005-05-28: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-28 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-29...\n",
      "Error fetching data for 2005-05-29: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-29 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-30...\n",
      "Error fetching data for 2005-05-30: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-30 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-05-31...\n",
      "Error fetching data for 2005-05-31: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-05-31 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-01...\n",
      "Error fetching data for 2005-06-01: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-01 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-02...\n",
      "Error fetching data for 2005-06-02: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-02 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-03...\n",
      "Error fetching data for 2005-06-03: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-03 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-04...\n",
      "Error fetching data for 2005-06-04: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-04 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-05...\n",
      "Error fetching data for 2005-06-05: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-05 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-06...\n",
      "Error fetching data for 2005-06-06: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-06 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-07...\n",
      "Error fetching data for 2005-06-07: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-07 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-08...\n",
      "Error fetching data for 2005-06-08: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-08 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-09...\n",
      "Error fetching data for 2005-06-09: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-09 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-10...\n",
      "Error fetching data for 2005-06-10: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-10 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-11...\n",
      "Error fetching data for 2005-06-11: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-11 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-12...\n",
      "Error fetching data for 2005-06-12: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-12 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-13...\n",
      "Error fetching data for 2005-06-13: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-13 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-14...\n",
      "Error fetching data for 2005-06-14: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-14 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-15...\n",
      "Error fetching data for 2005-06-15: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-15 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-16...\n",
      "Error fetching data for 2005-06-16: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-16 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-17...\n",
      "Error fetching data for 2005-06-17: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-17 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-18...\n",
      "Error fetching data for 2005-06-18: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-18 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-19...\n",
      "Error fetching data for 2005-06-19: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-19 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-20...\n",
      "Error fetching data for 2005-06-20: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-20 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-21...\n",
      "Error fetching data for 2005-06-21: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-21 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-22...\n",
      "Error fetching data for 2005-06-22: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-22 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-23...\n",
      "Error fetching data for 2005-06-23: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-23 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-24...\n",
      "Error fetching data for 2005-06-24: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-24 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-25...\n",
      "Error fetching data for 2005-06-25: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-25 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-26...\n",
      "Error fetching data for 2005-06-26: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-26 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-27...\n",
      "Error fetching data for 2005-06-27: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-27 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-28...\n",
      "Error fetching data for 2005-06-28: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-28 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-29...\n",
      "Error fetching data for 2005-06-29: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-29 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-06-30...\n",
      "Error fetching data for 2005-06-30: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-06-30 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-01...\n",
      "Error fetching data for 2005-07-01: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-01 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-02...\n",
      "Error fetching data for 2005-07-02: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-02 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-03...\n",
      "Error fetching data for 2005-07-03: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-03 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-04...\n",
      "Error fetching data for 2005-07-04: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-04 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-05...\n",
      "Error fetching data for 2005-07-05: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-05 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-06...\n",
      "Error fetching data for 2005-07-06: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-06 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-07...\n",
      "Error fetching data for 2005-07-07: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-07 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-08...\n",
      "Error fetching data for 2005-07-08: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-08 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-09...\n",
      "Error fetching data for 2005-07-09: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-09 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-10...\n",
      "Error fetching data for 2005-07-10: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-10 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-11...\n",
      "Error fetching data for 2005-07-11: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-11 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-12...\n",
      "Error fetching data for 2005-07-12: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-12 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-13...\n",
      "Error fetching data for 2005-07-13: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-13 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-14...\n",
      "Error fetching data for 2005-07-14: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-14 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-15...\n",
      "Error fetching data for 2005-07-15: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-15 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-16...\n",
      "Error fetching data for 2005-07-16: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-16 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-17...\n",
      "Error fetching data for 2005-07-17: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-17 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-18...\n",
      "Error fetching data for 2005-07-18: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-18 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-19...\n",
      "Error fetching data for 2005-07-19: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-19 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-20...\n",
      "Error fetching data for 2005-07-20: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-20 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-21...\n",
      "Error fetching data for 2005-07-21: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-21 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-22...\n",
      "Error fetching data for 2005-07-22: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-22 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-23...\n",
      "Error fetching data for 2005-07-23: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-23 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-24...\n",
      "Error fetching data for 2005-07-24: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-24 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-25...\n",
      "Error fetching data for 2005-07-25: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-25 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-26...\n",
      "Error fetching data for 2005-07-26: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-26 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-27...\n",
      "Error fetching data for 2005-07-27: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-27 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-28...\n",
      "Error fetching data for 2005-07-28: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-28 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-29...\n",
      "Error fetching data for 2005-07-29: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-29 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-30...\n",
      "Error fetching data for 2005-07-30: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-30 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-07-31...\n",
      "Error fetching data for 2005-07-31: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-07-31 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-01...\n",
      "Error fetching data for 2005-08-01: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-01 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-02...\n",
      "Error fetching data for 2005-08-02: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-02 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-03...\n",
      "Error fetching data for 2005-08-03: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-03 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-04...\n",
      "Error fetching data for 2005-08-04: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-04 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-05...\n",
      "Error fetching data for 2005-08-05: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-05 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-06...\n",
      "Error fetching data for 2005-08-06: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-06 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-07...\n",
      "Error fetching data for 2005-08-07: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-07 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-08...\n",
      "Error fetching data for 2005-08-08: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-08 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-09...\n",
      "Error fetching data for 2005-08-09: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-09 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-10...\n",
      "Error fetching data for 2005-08-10: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-10 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-11...\n",
      "Error fetching data for 2005-08-11: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-11 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-12...\n",
      "Error fetching data for 2005-08-12: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-12 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-13...\n",
      "Error fetching data for 2005-08-13: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-13 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-14...\n",
      "Error fetching data for 2005-08-14: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-14 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-15...\n",
      "Error fetching data for 2005-08-15: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-15 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-16...\n",
      "Error fetching data for 2005-08-16: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-16 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-17...\n",
      "Error fetching data for 2005-08-17: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-17 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-18...\n",
      "Error fetching data for 2005-08-18: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-18 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-19...\n",
      "Error fetching data for 2005-08-19: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-19 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-20...\n",
      "Error fetching data for 2005-08-20: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-20 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-21...\n",
      "Error fetching data for 2005-08-21: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-21 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-22...\n",
      "Error fetching data for 2005-08-22: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-22 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-23...\n",
      "Error fetching data for 2005-08-23: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-23 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-24...\n",
      "Error fetching data for 2005-08-24: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-24 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-25...\n",
      "Error fetching data for 2005-08-25: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-25 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-26...\n",
      "Error fetching data for 2005-08-26: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-26 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-27...\n",
      "Error fetching data for 2005-08-27: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-27 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-28...\n",
      "Error fetching data for 2005-08-28: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-28 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-29...\n",
      "Error fetching data for 2005-08-29: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-29 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-30...\n",
      "Error fetching data for 2005-08-30: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-30 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-08-31...\n",
      "Error fetching data for 2005-08-31: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-08-31 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-01...\n",
      "Error fetching data for 2005-09-01: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-01 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-02...\n",
      "Error fetching data for 2005-09-02: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-02 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-03...\n",
      "Error fetching data for 2005-09-03: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-03 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-04...\n",
      "Error fetching data for 2005-09-04: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-04 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-05...\n",
      "Error fetching data for 2005-09-05: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-05 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-06...\n",
      "Error fetching data for 2005-09-06: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-06 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-07...\n",
      "Error fetching data for 2005-09-07: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-07 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-08...\n",
      "Error fetching data for 2005-09-08: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-08 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-09...\n",
      "Error fetching data for 2005-09-09: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-09 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-10...\n",
      "Error fetching data for 2005-09-10: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-10 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-11...\n",
      "Error fetching data for 2005-09-11: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-11 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-12...\n",
      "Error fetching data for 2005-09-12: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-12 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-13...\n",
      "Error fetching data for 2005-09-13: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-13 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-14...\n",
      "Error fetching data for 2005-09-14: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-14 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-15...\n",
      "Error fetching data for 2005-09-15: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-15 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-16...\n",
      "Error fetching data for 2005-09-16: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-16 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-17...\n",
      "Error fetching data for 2005-09-17: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-17 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-18...\n",
      "Error fetching data for 2005-09-18: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-18 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-19...\n",
      "Error fetching data for 2005-09-19: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-19 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-20...\n",
      "Error fetching data for 2005-09-20: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-20 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-21...\n",
      "Error fetching data for 2005-09-21: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-21 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-22...\n",
      "Error fetching data for 2005-09-22: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-22 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-23...\n",
      "Error fetching data for 2005-09-23: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-23 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-24...\n",
      "Error fetching data for 2005-09-24: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-24 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-25...\n",
      "Error fetching data for 2005-09-25: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-25 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-26...\n",
      "Error fetching data for 2005-09-26: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-26 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-27...\n",
      "Error fetching data for 2005-09-27: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-27 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-28...\n",
      "Error fetching data for 2005-09-28: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-28 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-29...\n",
      "Error fetching data for 2005-09-29: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-29 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-09-30...\n",
      "Error fetching data for 2005-09-30: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-09-30 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-01...\n",
      "Error fetching data for 2005-10-01: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-01 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-02...\n",
      "Error fetching data for 2005-10-02: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-02 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-03...\n",
      "Error fetching data for 2005-10-03: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-03 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-04...\n",
      "Error fetching data for 2005-10-04: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-04 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-05...\n",
      "Error fetching data for 2005-10-05: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-05 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-06...\n",
      "Error fetching data for 2005-10-06: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-06 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-07...\n",
      "Error fetching data for 2005-10-07: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-07 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-08...\n",
      "Error fetching data for 2005-10-08: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-08 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-09...\n",
      "Error fetching data for 2005-10-09: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-09 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-10...\n",
      "Error fetching data for 2005-10-10: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-10 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-11...\n",
      "Error fetching data for 2005-10-11: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-11 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-12...\n",
      "Error fetching data for 2005-10-12: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-12 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-13...\n",
      "Error fetching data for 2005-10-13: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-13 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-14...\n",
      "Error fetching data for 2005-10-14: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-14 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-15...\n",
      "Error fetching data for 2005-10-15: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-15 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-16...\n",
      "Error fetching data for 2005-10-16: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-16 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-17...\n",
      "Error fetching data for 2005-10-17: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-17 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-18...\n",
      "Error fetching data for 2005-10-18: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-18 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-19...\n",
      "Error fetching data for 2005-10-19: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-19 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-20...\n",
      "Error fetching data for 2005-10-20: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-20 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-21...\n",
      "Error fetching data for 2005-10-21: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-21 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-22...\n",
      "Error fetching data for 2005-10-22: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-22 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-23...\n",
      "Error fetching data for 2005-10-23: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-23 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-24...\n",
      "Error fetching data for 2005-10-24: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-24 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-25...\n",
      "Error fetching data for 2005-10-25: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-25 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-26...\n",
      "Error fetching data for 2005-10-26: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-26 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-27...\n",
      "Error fetching data for 2005-10-27: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-27 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-28...\n",
      "Error fetching data for 2005-10-28: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-28 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-29...\n",
      "Error fetching data for 2005-10-29: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-29 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-30...\n",
      "Error fetching data for 2005-10-30: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-30 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-10-31...\n",
      "Error fetching data for 2005-10-31: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-10-31 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-01...\n",
      "Error fetching data for 2005-11-01: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-01 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-02...\n",
      "Error fetching data for 2005-11-02: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-02 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-03...\n",
      "Error fetching data for 2005-11-03: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-03 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-04...\n",
      "Error fetching data for 2005-11-04: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-04 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-05...\n",
      "Error fetching data for 2005-11-05: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-05 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-06...\n",
      "Error fetching data for 2005-11-06: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-06 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-07...\n",
      "Error fetching data for 2005-11-07: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-07 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-08...\n",
      "Error fetching data for 2005-11-08: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-08 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-09...\n",
      "Error fetching data for 2005-11-09: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-09 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-10...\n",
      "Error fetching data for 2005-11-10: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-10 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-11...\n",
      "Error fetching data for 2005-11-11: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-11 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-12...\n",
      "Error fetching data for 2005-11-12: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-12 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-13...\n",
      "Error fetching data for 2005-11-13: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-13 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-14...\n",
      "Error fetching data for 2005-11-14: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-14 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-15...\n",
      "Error fetching data for 2005-11-15: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-15 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-16...\n",
      "Error fetching data for 2005-11-16: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-16 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-17...\n",
      "Error fetching data for 2005-11-17: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-17 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-18...\n",
      "Error fetching data for 2005-11-18: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-18 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-19...\n",
      "Error fetching data for 2005-11-19: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-19 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-20...\n",
      "Error fetching data for 2005-11-20: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-20 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-21...\n",
      "Error fetching data for 2005-11-21: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-21 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-22...\n",
      "Error fetching data for 2005-11-22: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-22 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-23...\n",
      "Error fetching data for 2005-11-23: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-23 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-24...\n",
      "Error fetching data for 2005-11-24: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-24 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-25...\n",
      "Error fetching data for 2005-11-25: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-25 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-26...\n",
      "Error fetching data for 2005-11-26: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-26 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-27...\n",
      "Error fetching data for 2005-11-27: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-27 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-28...\n",
      "Error fetching data for 2005-11-28: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-28 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-29...\n",
      "Error fetching data for 2005-11-29: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-29 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-11-30...\n",
      "Error fetching data for 2005-11-30: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-11-30 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-01...\n",
      "Error fetching data for 2005-12-01: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-01 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-02...\n",
      "Error fetching data for 2005-12-02: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-02 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-03...\n",
      "Error fetching data for 2005-12-03: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-03 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-04...\n",
      "Error fetching data for 2005-12-04: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-04 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-05...\n",
      "Error fetching data for 2005-12-05: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-05 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-06...\n",
      "Error fetching data for 2005-12-06: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-06 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-07...\n",
      "Error fetching data for 2005-12-07: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-07 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-08...\n",
      "Error fetching data for 2005-12-08: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-08 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-09...\n",
      "Error fetching data for 2005-12-09: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-09 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-10...\n",
      "Error fetching data for 2005-12-10: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-10 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-11...\n",
      "Error fetching data for 2005-12-11: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-11 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-12...\n",
      "Error fetching data for 2005-12-12: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-12 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-13...\n",
      "Error fetching data for 2005-12-13: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-13 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-14...\n",
      "Error fetching data for 2005-12-14: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-14 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-15...\n",
      "Error fetching data for 2005-12-15: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-15 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-16...\n",
      "Error fetching data for 2005-12-16: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-16 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-17...\n",
      "Error fetching data for 2005-12-17: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-17 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-18...\n",
      "Error fetching data for 2005-12-18: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-18 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-19...\n",
      "Error fetching data for 2005-12-19: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-19 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-20...\n",
      "Error fetching data for 2005-12-20: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-20 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-21...\n",
      "Error fetching data for 2005-12-21: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-21 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-22...\n",
      "Error fetching data for 2005-12-22: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-22 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-23...\n",
      "Error fetching data for 2005-12-23: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-23 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-24...\n",
      "Error fetching data for 2005-12-24: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-24 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-25...\n",
      "Error fetching data for 2005-12-25: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-25 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-26...\n",
      "Error fetching data for 2005-12-26: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-26 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-27...\n",
      "Error fetching data for 2005-12-27: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-27 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-28...\n",
      "Error fetching data for 2005-12-28: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-28 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-29...\n",
      "Error fetching data for 2005-12-29: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-29 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-30...\n",
      "Error fetching data for 2005-12-30: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-30 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-12-31...\n",
      "Error fetching data for 2005-12-31: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-12-31 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-01...\n",
      "Error fetching data for 2006-01-01: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-01 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-02...\n",
      "Error fetching data for 2006-01-02: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-02 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-03...\n",
      "Error fetching data for 2006-01-03: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-03 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-04...\n",
      "Error fetching data for 2006-01-04: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-04 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-05...\n",
      "Error fetching data for 2006-01-05: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-05 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-06...\n",
      "Error fetching data for 2006-01-06: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-06 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-07...\n",
      "Error fetching data for 2006-01-07: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-07 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-08...\n",
      "Error fetching data for 2006-01-08: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-08 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-09...\n",
      "Error fetching data for 2006-01-09: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-09 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-10...\n",
      "Error fetching data for 2006-01-10: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-10 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-11...\n",
      "Error fetching data for 2006-01-11: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-11 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-12...\n",
      "Error fetching data for 2006-01-12: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-12 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-13...\n",
      "Error fetching data for 2006-01-13: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-13 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-14...\n",
      "Error fetching data for 2006-01-14: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-14 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-15...\n",
      "Error fetching data for 2006-01-15: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-15 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-16...\n",
      "Error fetching data for 2006-01-16: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-16 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-17...\n",
      "Error fetching data for 2006-01-17: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-17 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-18...\n",
      "Error fetching data for 2006-01-18: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-18 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-19...\n",
      "Error fetching data for 2006-01-19: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-19 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-20...\n",
      "Error fetching data for 2006-01-20: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-20 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-21...\n",
      "Error fetching data for 2006-01-21: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-21 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-22...\n",
      "Error fetching data for 2006-01-22: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-22 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-23...\n",
      "Error fetching data for 2006-01-23: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-23 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-24...\n",
      "Error fetching data for 2006-01-24: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-24 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-25...\n",
      "Error fetching data for 2006-01-25: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-25 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-26...\n",
      "Error fetching data for 2006-01-26: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-26 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-27...\n",
      "Error fetching data for 2006-01-27: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-27 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-28...\n",
      "Error fetching data for 2006-01-28: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-28 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-29...\n",
      "Error fetching data for 2006-01-29: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-29 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-30...\n",
      "Error fetching data for 2006-01-30: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-30 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-01-31...\n",
      "Error fetching data for 2006-01-31: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-01-31 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-01...\n",
      "Error fetching data for 2006-02-01: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-01 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-02...\n",
      "Error fetching data for 2006-02-02: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-02 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-03...\n",
      "Error fetching data for 2006-02-03: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-03 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-04...\n",
      "Error fetching data for 2006-02-04: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-04 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-05...\n",
      "Error fetching data for 2006-02-05: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-05 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-06...\n",
      "Error fetching data for 2006-02-06: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-06 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-07...\n",
      "Error fetching data for 2006-02-07: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-07 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-08...\n",
      "Error fetching data for 2006-02-08: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-08 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-09...\n",
      "Error fetching data for 2006-02-09: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-09 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-10...\n",
      "Error fetching data for 2006-02-10: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-10 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-11...\n",
      "Error fetching data for 2006-02-11: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-11 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-12...\n",
      "Error fetching data for 2006-02-12: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-12 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-13...\n",
      "Error fetching data for 2006-02-13: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-13 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-14...\n",
      "Error fetching data for 2006-02-14: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-14 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-15...\n",
      "Error fetching data for 2006-02-15: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-15 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-16...\n",
      "Error fetching data for 2006-02-16: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-16 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-17...\n",
      "Error fetching data for 2006-02-17: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-17 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-18...\n",
      "Error fetching data for 2006-02-18: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2006-02-18 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2006-02-19...\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[1], line 55\u001b[0m\n\u001b[0;32m     53\u001b[0m start_date \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m2005-02-01\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m     54\u001b[0m end_date \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m2024-12-10\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m---> 55\u001b[0m data \u001b[38;5;241m=\u001b[39m fetch_hourly_data(start_date, end_date)\n",
      "Cell \u001b[1;32mIn[1], line 30\u001b[0m, in \u001b[0;36mfetch_hourly_data\u001b[1;34m(start_date, end_date)\u001b[0m\n\u001b[0;32m     27\u001b[0m url \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mBASE_URL\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m?date=\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mdate_str\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m  \u001b[38;5;66;03m# Adjust based on the API's query parameter structure\u001b[39;00m\n\u001b[0;32m     29\u001b[0m \u001b[38;5;66;03m# Make the API request\u001b[39;00m\n\u001b[1;32m---> 30\u001b[0m response \u001b[38;5;241m=\u001b[39m requests\u001b[38;5;241m.\u001b[39mget(url)\n\u001b[0;32m     31\u001b[0m response\u001b[38;5;241m.\u001b[39mraise_for_status()\n\u001b[0;32m     32\u001b[0m daily_data \u001b[38;5;241m=\u001b[39m response\u001b[38;5;241m.\u001b[39mjson()  \u001b[38;5;66;03m# Assuming JSON response\u001b[39;00m\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\requests\\api.py:73\u001b[0m, in \u001b[0;36mget\u001b[1;34m(url, params, **kwargs)\u001b[0m\n\u001b[0;32m     62\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mget\u001b[39m(url, params\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[0;32m     63\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124;03m\"\"\"Sends a GET request.\u001b[39;00m\n\u001b[0;32m     64\u001b[0m \n\u001b[0;32m     65\u001b[0m \u001b[38;5;124;03m    :param url: URL for the new :class:`Request` object.\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m     70\u001b[0m \u001b[38;5;124;03m    :rtype: requests.Response\u001b[39;00m\n\u001b[0;32m     71\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[1;32m---> 73\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m request(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mget\u001b[39m\u001b[38;5;124m\"\u001b[39m, url, params\u001b[38;5;241m=\u001b[39mparams, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\requests\\api.py:59\u001b[0m, in \u001b[0;36mrequest\u001b[1;34m(method, url, **kwargs)\u001b[0m\n\u001b[0;32m     55\u001b[0m \u001b[38;5;66;03m# By using the 'with' statement we are sure the session is closed, thus we\u001b[39;00m\n\u001b[0;32m     56\u001b[0m \u001b[38;5;66;03m# avoid leaving sockets open which can trigger a ResourceWarning in some\u001b[39;00m\n\u001b[0;32m     57\u001b[0m \u001b[38;5;66;03m# cases, and look like a memory leak in others.\u001b[39;00m\n\u001b[0;32m     58\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m sessions\u001b[38;5;241m.\u001b[39mSession() \u001b[38;5;28;01mas\u001b[39;00m session:\n\u001b[1;32m---> 59\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m session\u001b[38;5;241m.\u001b[39mrequest(method\u001b[38;5;241m=\u001b[39mmethod, url\u001b[38;5;241m=\u001b[39murl, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\requests\\sessions.py:589\u001b[0m, in \u001b[0;36mSession.request\u001b[1;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[0;32m    584\u001b[0m send_kwargs \u001b[38;5;241m=\u001b[39m {\n\u001b[0;32m    585\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtimeout\u001b[39m\u001b[38;5;124m\"\u001b[39m: timeout,\n\u001b[0;32m    586\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mallow_redirects\u001b[39m\u001b[38;5;124m\"\u001b[39m: allow_redirects,\n\u001b[0;32m    587\u001b[0m }\n\u001b[0;32m    588\u001b[0m send_kwargs\u001b[38;5;241m.\u001b[39mupdate(settings)\n\u001b[1;32m--> 589\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msend(prep, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39msend_kwargs)\n\u001b[0;32m    591\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m resp\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\requests\\sessions.py:703\u001b[0m, in \u001b[0;36mSession.send\u001b[1;34m(self, request, **kwargs)\u001b[0m\n\u001b[0;32m    700\u001b[0m start \u001b[38;5;241m=\u001b[39m preferred_clock()\n\u001b[0;32m    702\u001b[0m \u001b[38;5;66;03m# Send the request\u001b[39;00m\n\u001b[1;32m--> 703\u001b[0m r \u001b[38;5;241m=\u001b[39m adapter\u001b[38;5;241m.\u001b[39msend(request, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[0;32m    705\u001b[0m \u001b[38;5;66;03m# Total elapsed time of the request (approximately)\u001b[39;00m\n\u001b[0;32m    706\u001b[0m elapsed \u001b[38;5;241m=\u001b[39m preferred_clock() \u001b[38;5;241m-\u001b[39m start\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\requests\\adapters.py:589\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[1;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[0;32m    586\u001b[0m     timeout \u001b[38;5;241m=\u001b[39m TimeoutSauce(connect\u001b[38;5;241m=\u001b[39mtimeout, read\u001b[38;5;241m=\u001b[39mtimeout)\n\u001b[0;32m    588\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 589\u001b[0m     resp \u001b[38;5;241m=\u001b[39m conn\u001b[38;5;241m.\u001b[39murlopen(\n\u001b[0;32m    590\u001b[0m         method\u001b[38;5;241m=\u001b[39mrequest\u001b[38;5;241m.\u001b[39mmethod,\n\u001b[0;32m    591\u001b[0m         url\u001b[38;5;241m=\u001b[39murl,\n\u001b[0;32m    592\u001b[0m         body\u001b[38;5;241m=\u001b[39mrequest\u001b[38;5;241m.\u001b[39mbody,\n\u001b[0;32m    593\u001b[0m         headers\u001b[38;5;241m=\u001b[39mrequest\u001b[38;5;241m.\u001b[39mheaders,\n\u001b[0;32m    594\u001b[0m         redirect\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[0;32m    595\u001b[0m         assert_same_host\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[0;32m    596\u001b[0m         preload_content\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[0;32m    597\u001b[0m         decode_content\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[0;32m    598\u001b[0m         retries\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmax_retries,\n\u001b[0;32m    599\u001b[0m         timeout\u001b[38;5;241m=\u001b[39mtimeout,\n\u001b[0;32m    600\u001b[0m         chunked\u001b[38;5;241m=\u001b[39mchunked,\n\u001b[0;32m    601\u001b[0m     )\n\u001b[0;32m    603\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (ProtocolError, \u001b[38;5;167;01mOSError\u001b[39;00m) \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[0;32m    604\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mConnectionError\u001b[39;00m(err, request\u001b[38;5;241m=\u001b[39mrequest)\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\urllib3\\connectionpool.py:789\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[1;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[0;32m    786\u001b[0m response_conn \u001b[38;5;241m=\u001b[39m conn \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m release_conn \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m    788\u001b[0m \u001b[38;5;66;03m# Make the request on the HTTPConnection object\u001b[39;00m\n\u001b[1;32m--> 789\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_make_request(\n\u001b[0;32m    790\u001b[0m     conn,\n\u001b[0;32m    791\u001b[0m     method,\n\u001b[0;32m    792\u001b[0m     url,\n\u001b[0;32m    793\u001b[0m     timeout\u001b[38;5;241m=\u001b[39mtimeout_obj,\n\u001b[0;32m    794\u001b[0m     body\u001b[38;5;241m=\u001b[39mbody,\n\u001b[0;32m    795\u001b[0m     headers\u001b[38;5;241m=\u001b[39mheaders,\n\u001b[0;32m    796\u001b[0m     chunked\u001b[38;5;241m=\u001b[39mchunked,\n\u001b[0;32m    797\u001b[0m     retries\u001b[38;5;241m=\u001b[39mretries,\n\u001b[0;32m    798\u001b[0m     response_conn\u001b[38;5;241m=\u001b[39mresponse_conn,\n\u001b[0;32m    799\u001b[0m     preload_content\u001b[38;5;241m=\u001b[39mpreload_content,\n\u001b[0;32m    800\u001b[0m     decode_content\u001b[38;5;241m=\u001b[39mdecode_content,\n\u001b[0;32m    801\u001b[0m     \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mresponse_kw,\n\u001b[0;32m    802\u001b[0m )\n\u001b[0;32m    804\u001b[0m \u001b[38;5;66;03m# Everything went great!\u001b[39;00m\n\u001b[0;32m    805\u001b[0m clean_exit \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\urllib3\\connectionpool.py:466\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[1;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001b[0m\n\u001b[0;32m    463\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m    464\u001b[0m     \u001b[38;5;66;03m# Trigger any extra validation we need to do.\u001b[39;00m\n\u001b[0;32m    465\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 466\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_validate_conn(conn)\n\u001b[0;32m    467\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m (SocketTimeout, BaseSSLError) \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[0;32m    468\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_raise_timeout(err\u001b[38;5;241m=\u001b[39me, url\u001b[38;5;241m=\u001b[39murl, timeout_value\u001b[38;5;241m=\u001b[39mconn\u001b[38;5;241m.\u001b[39mtimeout)\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\urllib3\\connectionpool.py:1095\u001b[0m, in \u001b[0;36mHTTPSConnectionPool._validate_conn\u001b[1;34m(self, conn)\u001b[0m\n\u001b[0;32m   1093\u001b[0m \u001b[38;5;66;03m# Force connect early to allow us to validate the connection.\u001b[39;00m\n\u001b[0;32m   1094\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mis_closed:\n\u001b[1;32m-> 1095\u001b[0m     conn\u001b[38;5;241m.\u001b[39mconnect()\n\u001b[0;32m   1097\u001b[0m \u001b[38;5;66;03m# TODO revise this, see https://github.com/urllib3/urllib3/issues/2791\u001b[39;00m\n\u001b[0;32m   1098\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mis_verified \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mproxy_is_verified:\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\urllib3\\connection.py:652\u001b[0m, in \u001b[0;36mHTTPSConnection.connect\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    649\u001b[0m \u001b[38;5;66;03m# Remove trailing '.' from fqdn hostnames to allow certificate validation\u001b[39;00m\n\u001b[0;32m    650\u001b[0m server_hostname_rm_dot \u001b[38;5;241m=\u001b[39m server_hostname\u001b[38;5;241m.\u001b[39mrstrip(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m.\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m--> 652\u001b[0m sock_and_verified \u001b[38;5;241m=\u001b[39m _ssl_wrap_socket_and_match_hostname(\n\u001b[0;32m    653\u001b[0m     sock\u001b[38;5;241m=\u001b[39msock,\n\u001b[0;32m    654\u001b[0m     cert_reqs\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcert_reqs,\n\u001b[0;32m    655\u001b[0m     ssl_version\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mssl_version,\n\u001b[0;32m    656\u001b[0m     ssl_minimum_version\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mssl_minimum_version,\n\u001b[0;32m    657\u001b[0m     ssl_maximum_version\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mssl_maximum_version,\n\u001b[0;32m    658\u001b[0m     ca_certs\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mca_certs,\n\u001b[0;32m    659\u001b[0m     ca_cert_dir\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mca_cert_dir,\n\u001b[0;32m    660\u001b[0m     ca_cert_data\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mca_cert_data,\n\u001b[0;32m    661\u001b[0m     cert_file\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcert_file,\n\u001b[0;32m    662\u001b[0m     key_file\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mkey_file,\n\u001b[0;32m    663\u001b[0m     key_password\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mkey_password,\n\u001b[0;32m    664\u001b[0m     server_hostname\u001b[38;5;241m=\u001b[39mserver_hostname_rm_dot,\n\u001b[0;32m    665\u001b[0m     ssl_context\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mssl_context,\n\u001b[0;32m    666\u001b[0m     tls_in_tls\u001b[38;5;241m=\u001b[39mtls_in_tls,\n\u001b[0;32m    667\u001b[0m     assert_hostname\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39massert_hostname,\n\u001b[0;32m    668\u001b[0m     assert_fingerprint\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39massert_fingerprint,\n\u001b[0;32m    669\u001b[0m )\n\u001b[0;32m    670\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msock \u001b[38;5;241m=\u001b[39m sock_and_verified\u001b[38;5;241m.\u001b[39msocket\n\u001b[0;32m    672\u001b[0m \u001b[38;5;66;03m# Forwarding proxies can never have a verified target since\u001b[39;00m\n\u001b[0;32m    673\u001b[0m \u001b[38;5;66;03m# the proxy is the one doing the verification. Should instead\u001b[39;00m\n\u001b[0;32m    674\u001b[0m \u001b[38;5;66;03m# use a CONNECT tunnel in order to verify the target.\u001b[39;00m\n\u001b[0;32m    675\u001b[0m \u001b[38;5;66;03m# See: https://github.com/urllib3/urllib3/issues/3267.\u001b[39;00m\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\urllib3\\connection.py:805\u001b[0m, in \u001b[0;36m_ssl_wrap_socket_and_match_hostname\u001b[1;34m(sock, cert_reqs, ssl_version, ssl_minimum_version, ssl_maximum_version, cert_file, key_file, key_password, ca_certs, ca_cert_dir, ca_cert_data, assert_hostname, assert_fingerprint, server_hostname, ssl_context, tls_in_tls)\u001b[0m\n\u001b[0;32m    802\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m is_ipaddress(normalized):\n\u001b[0;32m    803\u001b[0m         server_hostname \u001b[38;5;241m=\u001b[39m normalized\n\u001b[1;32m--> 805\u001b[0m ssl_sock \u001b[38;5;241m=\u001b[39m ssl_wrap_socket(\n\u001b[0;32m    806\u001b[0m     sock\u001b[38;5;241m=\u001b[39msock,\n\u001b[0;32m    807\u001b[0m     keyfile\u001b[38;5;241m=\u001b[39mkey_file,\n\u001b[0;32m    808\u001b[0m     certfile\u001b[38;5;241m=\u001b[39mcert_file,\n\u001b[0;32m    809\u001b[0m     key_password\u001b[38;5;241m=\u001b[39mkey_password,\n\u001b[0;32m    810\u001b[0m     ca_certs\u001b[38;5;241m=\u001b[39mca_certs,\n\u001b[0;32m    811\u001b[0m     ca_cert_dir\u001b[38;5;241m=\u001b[39mca_cert_dir,\n\u001b[0;32m    812\u001b[0m     ca_cert_data\u001b[38;5;241m=\u001b[39mca_cert_data,\n\u001b[0;32m    813\u001b[0m     server_hostname\u001b[38;5;241m=\u001b[39mserver_hostname,\n\u001b[0;32m    814\u001b[0m     ssl_context\u001b[38;5;241m=\u001b[39mcontext,\n\u001b[0;32m    815\u001b[0m     tls_in_tls\u001b[38;5;241m=\u001b[39mtls_in_tls,\n\u001b[0;32m    816\u001b[0m )\n\u001b[0;32m    818\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m    819\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m assert_fingerprint:\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\urllib3\\util\\ssl_.py:465\u001b[0m, in \u001b[0;36mssl_wrap_socket\u001b[1;34m(sock, keyfile, certfile, cert_reqs, ca_certs, server_hostname, ssl_version, ciphers, ssl_context, ca_cert_dir, key_password, ca_cert_data, tls_in_tls)\u001b[0m\n\u001b[0;32m    462\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mNotImplementedError\u001b[39;00m:  \u001b[38;5;66;03m# Defensive: in CI, we always have set_alpn_protocols\u001b[39;00m\n\u001b[0;32m    463\u001b[0m     \u001b[38;5;28;01mpass\u001b[39;00m\n\u001b[1;32m--> 465\u001b[0m ssl_sock \u001b[38;5;241m=\u001b[39m _ssl_wrap_socket_impl(sock, context, tls_in_tls, server_hostname)\n\u001b[0;32m    466\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m ssl_sock\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\urllib3\\util\\ssl_.py:509\u001b[0m, in \u001b[0;36m_ssl_wrap_socket_impl\u001b[1;34m(sock, ssl_context, tls_in_tls, server_hostname)\u001b[0m\n\u001b[0;32m    506\u001b[0m     SSLTransport\u001b[38;5;241m.\u001b[39m_validate_ssl_context_for_tls_in_tls(ssl_context)\n\u001b[0;32m    507\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m SSLTransport(sock, ssl_context, server_hostname)\n\u001b[1;32m--> 509\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m ssl_context\u001b[38;5;241m.\u001b[39mwrap_socket(sock, server_hostname\u001b[38;5;241m=\u001b[39mserver_hostname)\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\ssl.py:455\u001b[0m, in \u001b[0;36mSSLContext.wrap_socket\u001b[1;34m(self, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, session)\u001b[0m\n\u001b[0;32m    449\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwrap_socket\u001b[39m(\u001b[38;5;28mself\u001b[39m, sock, server_side\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[0;32m    450\u001b[0m                 do_handshake_on_connect\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[0;32m    451\u001b[0m                 suppress_ragged_eofs\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[0;32m    452\u001b[0m                 server_hostname\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, session\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[0;32m    453\u001b[0m     \u001b[38;5;66;03m# SSLSocket class handles server_hostname encoding before it calls\u001b[39;00m\n\u001b[0;32m    454\u001b[0m     \u001b[38;5;66;03m# ctx._wrap_socket()\u001b[39;00m\n\u001b[1;32m--> 455\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msslsocket_class\u001b[38;5;241m.\u001b[39m_create(\n\u001b[0;32m    456\u001b[0m         sock\u001b[38;5;241m=\u001b[39msock,\n\u001b[0;32m    457\u001b[0m         server_side\u001b[38;5;241m=\u001b[39mserver_side,\n\u001b[0;32m    458\u001b[0m         do_handshake_on_connect\u001b[38;5;241m=\u001b[39mdo_handshake_on_connect,\n\u001b[0;32m    459\u001b[0m         suppress_ragged_eofs\u001b[38;5;241m=\u001b[39msuppress_ragged_eofs,\n\u001b[0;32m    460\u001b[0m         server_hostname\u001b[38;5;241m=\u001b[39mserver_hostname,\n\u001b[0;32m    461\u001b[0m         context\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m,\n\u001b[0;32m    462\u001b[0m         session\u001b[38;5;241m=\u001b[39msession\n\u001b[0;32m    463\u001b[0m     )\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\ssl.py:1042\u001b[0m, in \u001b[0;36mSSLSocket._create\u001b[1;34m(cls, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, context, session)\u001b[0m\n\u001b[0;32m   1039\u001b[0m             \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0.0\u001b[39m:\n\u001b[0;32m   1040\u001b[0m                 \u001b[38;5;66;03m# non-blocking\u001b[39;00m\n\u001b[0;32m   1041\u001b[0m                 \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdo_handshake_on_connect should not be specified for non-blocking sockets\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m-> 1042\u001b[0m             \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdo_handshake()\n\u001b[0;32m   1043\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m:\n\u001b[0;32m   1044\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\ssl.py:1320\u001b[0m, in \u001b[0;36mSSLSocket.do_handshake\u001b[1;34m(self, block)\u001b[0m\n\u001b[0;32m   1318\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0.0\u001b[39m \u001b[38;5;129;01mand\u001b[39;00m block:\n\u001b[0;32m   1319\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msettimeout(\u001b[38;5;28;01mNone\u001b[39;00m)\n\u001b[1;32m-> 1320\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_sslobj\u001b[38;5;241m.\u001b[39mdo_handshake()\n\u001b[0;32m   1321\u001b[0m \u001b[38;5;28;01mfinally\u001b[39;00m:\n\u001b[0;32m   1322\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msettimeout(timeout)\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import requests\n",
    "import pandas as pd\n",
    "from datetime import datetime, timedelta\n",
    "\n",
    "# Define the base URL for the API or dataset\n",
    "BASE_URL = \"https://api.nyiso.com/load_data\"  # Replace this with the actual endpoint\n",
    "\n",
    "# Function to fetch data from the API\n",
    "def fetch_hourly_data(start_date, end_date):\n",
    "    \"\"\"\n",
    "    Fetch hourly data from the API within the given date range.\n",
    "    :param start_date: Start date (YYYY-MM-DD)\n",
    "    :param end_date: End date (YYYY-MM-DD)\n",
    "    :return: DataFrame with hourly data\n",
    "    \"\"\"\n",
    "    data = []\n",
    "    current_date = datetime.strptime(start_date, \"%Y-%m-%d\")\n",
    "    end_date = datetime.strptime(end_date, \"%Y-%m-%d\")\n",
    "    \n",
    "    while current_date <= end_date:\n",
    "        try:\n",
    "            # Format the date for the API request\n",
    "            date_str = current_date.strftime(\"%Y-%m-%d\")\n",
    "            print(f\"Fetching data for {date_str}...\")\n",
    "            \n",
    "            # Build the URL dynamically based on the date\n",
    "            url = f\"{BASE_URL}?date={date_str}\"  # Adjust based on the API's query parameter structure\n",
    "            \n",
    "            # Make the API request\n",
    "            response = requests.get(url)\n",
    "            response.raise_for_status()\n",
    "            daily_data = response.json()  # Assuming JSON response\n",
    "            \n",
    "            # Process the data (adjust based on API response structure)\n",
    "            for record in daily_data[\"hourly_load\"]:  # Update this based on actual keys in the API response\n",
    "                data.append({\n",
    "                    \"County\": record[\"county\"],  # Adjust key names\n",
    "                    \"Date\": record[\"date\"],  # Adjust key names\n",
    "                    \"Hour\": record[\"hour\"],  # Adjust key names\n",
    "                    \"Consumption (MWh)\": record[\"load\"]  # Adjust key names\n",
    "                })\n",
    "                \n",
    "        except Exception as e:\n",
    "            print(f\"Error fetching data for {date_str}: {e}\")\n",
    "        \n",
    "        # Move to the next day\n",
    "        current_date += timedelta(days=1)\n",
    "    \n",
    "    # Convert to DataFrame\n",
    "    return pd.DataFrame(data)\n",
    "\n",
    "# Fetch and save data\n",
    "start_date = \"2005-02-01\"\n",
    "end_date = \"2024-12-10\"\n",
    "data = fetch_hourly_data(start_date, end_date)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a5c089d5-aa3a-49a1-ae0c-2edd69f9adc5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fetching data for 2005-02-01...\n",
      "Error fetching data for 2005-02-01: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-01 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-02...\n",
      "Error fetching data for 2005-02-02: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-02 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-03...\n",
      "Error fetching data for 2005-02-03: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-03 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-04...\n",
      "Error fetching data for 2005-02-04: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-04 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-05...\n",
      "Error fetching data for 2005-02-05: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-05 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-06...\n",
      "Error fetching data for 2005-02-06: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-06 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-07...\n",
      "Error fetching data for 2005-02-07: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-07 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-08...\n",
      "Error fetching data for 2005-02-08: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-08 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-09...\n",
      "Error fetching data for 2005-02-09: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-09 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-10...\n",
      "Error fetching data for 2005-02-10: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-10 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-11...\n",
      "Error fetching data for 2005-02-11: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-11 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-12...\n",
      "Error fetching data for 2005-02-12: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-12 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-13...\n",
      "Error fetching data for 2005-02-13: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-13 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-14...\n",
      "Error fetching data for 2005-02-14: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-14 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-15...\n",
      "Error fetching data for 2005-02-15: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-15 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-16...\n",
      "Error fetching data for 2005-02-16: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-16 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-17...\n",
      "Error fetching data for 2005-02-17: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-17 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-18...\n",
      "Error fetching data for 2005-02-18: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-18 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-19...\n",
      "Error fetching data for 2005-02-19: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-19 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-20...\n",
      "Error fetching data for 2005-02-20: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-20 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-21...\n",
      "Error fetching data for 2005-02-21: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-21 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-22...\n",
      "Error fetching data for 2005-02-22: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-22 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-23...\n",
      "Error fetching data for 2005-02-23: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-23 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-24...\n",
      "Error fetching data for 2005-02-24: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-24 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-25...\n",
      "Error fetching data for 2005-02-25: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-25 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-26...\n",
      "Error fetching data for 2005-02-26: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-26 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-27...\n",
      "Error fetching data for 2005-02-27: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-27 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-02-28...\n",
      "Error fetching data for 2005-02-28: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-02-28 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-01...\n",
      "Error fetching data for 2005-03-01: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-01 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-02...\n",
      "Error fetching data for 2005-03-02: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-02 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-03...\n",
      "Error fetching data for 2005-03-03: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-03 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-04...\n",
      "Error fetching data for 2005-03-04: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-04 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-05...\n",
      "Error fetching data for 2005-03-05: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-05 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-06...\n",
      "Error fetching data for 2005-03-06: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-06 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-07...\n",
      "Error fetching data for 2005-03-07: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-07 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-08...\n",
      "Error fetching data for 2005-03-08: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-08 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-09...\n",
      "Error fetching data for 2005-03-09: HTTPSConnectionPool(host='api.nyiso.com', port=443): Max retries exceeded with url: /load_data?date=2005-03-09 (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)')))\n",
      "Fetching data for 2005-03-10...\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[2], line 49\u001b[0m\n\u001b[0;32m     46\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m pd\u001b[38;5;241m.\u001b[39mDataFrame(data)\n\u001b[0;32m     48\u001b[0m \u001b[38;5;66;03m# Fetch data\u001b[39;00m\n\u001b[1;32m---> 49\u001b[0m data \u001b[38;5;241m=\u001b[39m fetch_hourly_data(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m2005-02-01\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m2024-12-10\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "Cell \u001b[1;32mIn[2], line 29\u001b[0m, in \u001b[0;36mfetch_hourly_data\u001b[1;34m(start_date, end_date)\u001b[0m\n\u001b[0;32m     27\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mFetching data for \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mdate_str\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m...\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m     28\u001b[0m url \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mBASE_URL\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m?date=\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mdate_str\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m---> 29\u001b[0m response \u001b[38;5;241m=\u001b[39m session\u001b[38;5;241m.\u001b[39mget(url)\n\u001b[0;32m     30\u001b[0m response\u001b[38;5;241m.\u001b[39mraise_for_status()\n\u001b[0;32m     31\u001b[0m daily_data \u001b[38;5;241m=\u001b[39m response\u001b[38;5;241m.\u001b[39mjson()\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\requests\\sessions.py:602\u001b[0m, in \u001b[0;36mSession.get\u001b[1;34m(self, url, **kwargs)\u001b[0m\n\u001b[0;32m    594\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124;03m\"\"\"Sends a GET request. Returns :class:`Response` object.\u001b[39;00m\n\u001b[0;32m    595\u001b[0m \n\u001b[0;32m    596\u001b[0m \u001b[38;5;124;03m:param url: URL for the new :class:`Request` object.\u001b[39;00m\n\u001b[0;32m    597\u001b[0m \u001b[38;5;124;03m:param \\*\\*kwargs: Optional arguments that ``request`` takes.\u001b[39;00m\n\u001b[0;32m    598\u001b[0m \u001b[38;5;124;03m:rtype: requests.Response\u001b[39;00m\n\u001b[0;32m    599\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[0;32m    601\u001b[0m kwargs\u001b[38;5;241m.\u001b[39msetdefault(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mallow_redirects\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[1;32m--> 602\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mrequest(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mGET\u001b[39m\u001b[38;5;124m\"\u001b[39m, url, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\requests\\sessions.py:589\u001b[0m, in \u001b[0;36mSession.request\u001b[1;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[0;32m    584\u001b[0m send_kwargs \u001b[38;5;241m=\u001b[39m {\n\u001b[0;32m    585\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtimeout\u001b[39m\u001b[38;5;124m\"\u001b[39m: timeout,\n\u001b[0;32m    586\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mallow_redirects\u001b[39m\u001b[38;5;124m\"\u001b[39m: allow_redirects,\n\u001b[0;32m    587\u001b[0m }\n\u001b[0;32m    588\u001b[0m send_kwargs\u001b[38;5;241m.\u001b[39mupdate(settings)\n\u001b[1;32m--> 589\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msend(prep, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39msend_kwargs)\n\u001b[0;32m    591\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m resp\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\requests\\sessions.py:703\u001b[0m, in \u001b[0;36mSession.send\u001b[1;34m(self, request, **kwargs)\u001b[0m\n\u001b[0;32m    700\u001b[0m start \u001b[38;5;241m=\u001b[39m preferred_clock()\n\u001b[0;32m    702\u001b[0m \u001b[38;5;66;03m# Send the request\u001b[39;00m\n\u001b[1;32m--> 703\u001b[0m r \u001b[38;5;241m=\u001b[39m adapter\u001b[38;5;241m.\u001b[39msend(request, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[0;32m    705\u001b[0m \u001b[38;5;66;03m# Total elapsed time of the request (approximately)\u001b[39;00m\n\u001b[0;32m    706\u001b[0m elapsed \u001b[38;5;241m=\u001b[39m preferred_clock() \u001b[38;5;241m-\u001b[39m start\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\requests\\adapters.py:589\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[1;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[0;32m    586\u001b[0m     timeout \u001b[38;5;241m=\u001b[39m TimeoutSauce(connect\u001b[38;5;241m=\u001b[39mtimeout, read\u001b[38;5;241m=\u001b[39mtimeout)\n\u001b[0;32m    588\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 589\u001b[0m     resp \u001b[38;5;241m=\u001b[39m conn\u001b[38;5;241m.\u001b[39murlopen(\n\u001b[0;32m    590\u001b[0m         method\u001b[38;5;241m=\u001b[39mrequest\u001b[38;5;241m.\u001b[39mmethod,\n\u001b[0;32m    591\u001b[0m         url\u001b[38;5;241m=\u001b[39murl,\n\u001b[0;32m    592\u001b[0m         body\u001b[38;5;241m=\u001b[39mrequest\u001b[38;5;241m.\u001b[39mbody,\n\u001b[0;32m    593\u001b[0m         headers\u001b[38;5;241m=\u001b[39mrequest\u001b[38;5;241m.\u001b[39mheaders,\n\u001b[0;32m    594\u001b[0m         redirect\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[0;32m    595\u001b[0m         assert_same_host\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[0;32m    596\u001b[0m         preload_content\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[0;32m    597\u001b[0m         decode_content\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[0;32m    598\u001b[0m         retries\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmax_retries,\n\u001b[0;32m    599\u001b[0m         timeout\u001b[38;5;241m=\u001b[39mtimeout,\n\u001b[0;32m    600\u001b[0m         chunked\u001b[38;5;241m=\u001b[39mchunked,\n\u001b[0;32m    601\u001b[0m     )\n\u001b[0;32m    603\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (ProtocolError, \u001b[38;5;167;01mOSError\u001b[39;00m) \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[0;32m    604\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mConnectionError\u001b[39;00m(err, request\u001b[38;5;241m=\u001b[39mrequest)\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\urllib3\\connectionpool.py:789\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[1;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)\u001b[0m\n\u001b[0;32m    786\u001b[0m response_conn \u001b[38;5;241m=\u001b[39m conn \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m release_conn \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m    788\u001b[0m \u001b[38;5;66;03m# Make the request on the HTTPConnection object\u001b[39;00m\n\u001b[1;32m--> 789\u001b[0m response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_make_request(\n\u001b[0;32m    790\u001b[0m     conn,\n\u001b[0;32m    791\u001b[0m     method,\n\u001b[0;32m    792\u001b[0m     url,\n\u001b[0;32m    793\u001b[0m     timeout\u001b[38;5;241m=\u001b[39mtimeout_obj,\n\u001b[0;32m    794\u001b[0m     body\u001b[38;5;241m=\u001b[39mbody,\n\u001b[0;32m    795\u001b[0m     headers\u001b[38;5;241m=\u001b[39mheaders,\n\u001b[0;32m    796\u001b[0m     chunked\u001b[38;5;241m=\u001b[39mchunked,\n\u001b[0;32m    797\u001b[0m     retries\u001b[38;5;241m=\u001b[39mretries,\n\u001b[0;32m    798\u001b[0m     response_conn\u001b[38;5;241m=\u001b[39mresponse_conn,\n\u001b[0;32m    799\u001b[0m     preload_content\u001b[38;5;241m=\u001b[39mpreload_content,\n\u001b[0;32m    800\u001b[0m     decode_content\u001b[38;5;241m=\u001b[39mdecode_content,\n\u001b[0;32m    801\u001b[0m     \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mresponse_kw,\n\u001b[0;32m    802\u001b[0m )\n\u001b[0;32m    804\u001b[0m \u001b[38;5;66;03m# Everything went great!\u001b[39;00m\n\u001b[0;32m    805\u001b[0m clean_exit \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\urllib3\\connectionpool.py:466\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[1;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)\u001b[0m\n\u001b[0;32m    463\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m    464\u001b[0m     \u001b[38;5;66;03m# Trigger any extra validation we need to do.\u001b[39;00m\n\u001b[0;32m    465\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 466\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_validate_conn(conn)\n\u001b[0;32m    467\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m (SocketTimeout, BaseSSLError) \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[0;32m    468\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_raise_timeout(err\u001b[38;5;241m=\u001b[39me, url\u001b[38;5;241m=\u001b[39murl, timeout_value\u001b[38;5;241m=\u001b[39mconn\u001b[38;5;241m.\u001b[39mtimeout)\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\urllib3\\connectionpool.py:1095\u001b[0m, in \u001b[0;36mHTTPSConnectionPool._validate_conn\u001b[1;34m(self, conn)\u001b[0m\n\u001b[0;32m   1093\u001b[0m \u001b[38;5;66;03m# Force connect early to allow us to validate the connection.\u001b[39;00m\n\u001b[0;32m   1094\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mis_closed:\n\u001b[1;32m-> 1095\u001b[0m     conn\u001b[38;5;241m.\u001b[39mconnect()\n\u001b[0;32m   1097\u001b[0m \u001b[38;5;66;03m# TODO revise this, see https://github.com/urllib3/urllib3/issues/2791\u001b[39;00m\n\u001b[0;32m   1098\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mis_verified \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mproxy_is_verified:\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\urllib3\\connection.py:615\u001b[0m, in \u001b[0;36mHTTPSConnection.connect\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    613\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mconnect\u001b[39m(\u001b[38;5;28mself\u001b[39m) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m    614\u001b[0m     sock: socket\u001b[38;5;241m.\u001b[39msocket \u001b[38;5;241m|\u001b[39m ssl\u001b[38;5;241m.\u001b[39mSSLSocket\n\u001b[1;32m--> 615\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msock \u001b[38;5;241m=\u001b[39m sock \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_new_conn()\n\u001b[0;32m    616\u001b[0m     server_hostname: \u001b[38;5;28mstr\u001b[39m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhost\n\u001b[0;32m    617\u001b[0m     tls_in_tls \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mFalse\u001b[39;00m\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\urllib3\\connection.py:196\u001b[0m, in \u001b[0;36mHTTPConnection._new_conn\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    191\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Establish a socket connection and set nodelay settings on it.\u001b[39;00m\n\u001b[0;32m    192\u001b[0m \n\u001b[0;32m    193\u001b[0m \u001b[38;5;124;03m:return: New socket connection.\u001b[39;00m\n\u001b[0;32m    194\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[0;32m    195\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 196\u001b[0m     sock \u001b[38;5;241m=\u001b[39m connection\u001b[38;5;241m.\u001b[39mcreate_connection(\n\u001b[0;32m    197\u001b[0m         (\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_dns_host, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mport),\n\u001b[0;32m    198\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtimeout,\n\u001b[0;32m    199\u001b[0m         source_address\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msource_address,\n\u001b[0;32m    200\u001b[0m         socket_options\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msocket_options,\n\u001b[0;32m    201\u001b[0m     )\n\u001b[0;32m    202\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m socket\u001b[38;5;241m.\u001b[39mgaierror \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[0;32m    203\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m NameResolutionError(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhost, \u001b[38;5;28mself\u001b[39m, e) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01me\u001b[39;00m\n",
      "File \u001b[1;32m~\\anaconda3\\Lib\\site-packages\\urllib3\\util\\connection.py:73\u001b[0m, in \u001b[0;36mcreate_connection\u001b[1;34m(address, timeout, source_address, socket_options)\u001b[0m\n\u001b[0;32m     71\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m source_address:\n\u001b[0;32m     72\u001b[0m     sock\u001b[38;5;241m.\u001b[39mbind(source_address)\n\u001b[1;32m---> 73\u001b[0m sock\u001b[38;5;241m.\u001b[39mconnect(sa)\n\u001b[0;32m     74\u001b[0m \u001b[38;5;66;03m# Break explicitly a reference cycle\u001b[39;00m\n\u001b[0;32m     75\u001b[0m err \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import requests\n",
    "from requests.adapters import HTTPAdapter\n",
    "from urllib3.util.ssl_ import create_urllib3_context\n",
    "\n",
    "class TLSAdapter(HTTPAdapter):\n",
    "    def __init__(self, tls_version=None, *args, **kwargs):\n",
    "        self.tls_version = tls_version\n",
    "        self.context = create_urllib3_context()\n",
    "        if tls_version:\n",
    "            self.context.minimum_version = tls_version\n",
    "        super().__init__(*args, **kwargs)\n",
    "\n",
    "# Create a session with a specific TLS version (e.g., TLS 1.2)\n",
    "session = requests.Session()\n",
    "adapter = TLSAdapter()\n",
    "session.mount(\"https://\", adapter)\n",
    "\n",
    "# Example modified fetch\n",
    "def fetch_hourly_data(start_date, end_date):\n",
    "    data = []\n",
    "    current_date = datetime.strptime(start_date, \"%Y-%m-%d\")\n",
    "    end_date = datetime.strptime(end_date, \"%Y-%m-%d\")\n",
    "    \n",
    "    while current_date <= end_date:\n",
    "        try:\n",
    "            date_str = current_date.strftime(\"%Y-%m-%d\")\n",
    "            print(f\"Fetching data for {date_str}...\")\n",
    "            url = f\"{BASE_URL}?date={date_str}\"\n",
    "            response = session.get(url)\n",
    "            response.raise_for_status()\n",
    "            daily_data = response.json()\n",
    "            \n",
    "            # Process data\n",
    "            for record in daily_data.get(\"hourly_load\", []):\n",
    "                data.append({\n",
    "                    \"County\": record.get(\"county\"),\n",
    "                    \"Date\": record.get(\"date\"),\n",
    "                    \"Hour\": record.get(\"hour\"),\n",
    "                    \"Consumption (MWh)\": record.get(\"load\")\n",
    "                })\n",
    "        except Exception as e:\n",
    "            print(f\"Error fetching data for {date_str}: {e}\")\n",
    "        \n",
    "        current_date += timedelta(days=1)\n",
    "    \n",
    "    return pd.DataFrame(data)\n",
    "\n",
    "# Fetch data\n",
    "data = fetch_hourly_data(\"2005-02-01\", \"2024-12-10\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "92f8acd3-2c0b-455e-95fe-39fc5c691865",
   "metadata": {},
   "outputs": [],
   "source": [
    "from requests.adapters import Retry\n",
    "\n",
    "retry_strategy = Retry(\n",
    "    total=5,\n",
    "    backoff_factor=1,\n",
    "    status_forcelist=[429, 500, 502, 503, 504]\n",
    ")\n",
    "adapter = HTTPAdapter(max_retries=retry_strategy)\n",
    "session.mount(\"https://\", adapter)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dc0279b5-cc55-4fce-b0e7-bb6c90bbba8e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
